### ISSUE TITLE
Literal value is not captured by TypeVar

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#1673]
The following code generates an error when it should not.

```python
from typing import Literal, TypeVar

Format = Literal["json", "jsonl", "csv", "text"]

_T = TypeVar("_T")

def foo(x: _T) -> _T:
    ...

format: Format = foo("text") # Error: Expression of type "str" cannot be assigned to declared type "Format"
```

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.124, which I just published. It will also be included in the next release of pylance.
